### ISSUE TITLE
bug: Ion-item-sliding option can be clicked for a moment after item is closed

### ISSUE THREAD

**Issue by mrSingh007** [ionic-team/ionic-framework#22722]
'ion-item-sliding' 'option' is clickable even when slide is closed. 

for example in the following is user click on same location where 'Lorem' button was displayed, after closing the slide, it will trigger 'unread' function. 
```
 <ion-item-sliding>
    <ion-item>
      <ion-label>Item Options</ion-label>
      <ion-icon name="trash" (click)="ipsum()"></ion-icon>
    </ion-item>
    <ion-item-options side="end">
      <ion-item-option (click)="unread(item)">Lorem</ion-item-option>
    </ion-item-options>
  </ion-item-sliding>
```

Problem is that after sliding is closed it takes bit time to remove 'slide-active' classes. In this mean time when user click on item on same function, it get clicked on item option even item option is not visible.

**Ionic version:**
ionic --version:  6.12.1

**Expected behavior:**
After item sliding is closed the options should not be clickable. This should happen fast.

**Steps to reproduce:**
 From above code, Just slide to see 'Lorem' option.
 Now close the slide and click on same position where lorem option was displayed. (or on trash icon)
    - it will trigger 'unread' function.

**Comment by mrSingh007** [comment]
@liamdebeasi 
Here is reproduction on Stackblitz   https://stackblitz.com/edit/angular-ivy-p5wn6z?file=src/app/app.component.html 
and video https://drive.google.com/file/d/1qiSioSmwA6CMNcDMP1uf6cKj--80GArs

**Comment by liamdebeasi** [comment]
Thanks, I can reproduce the issue. Looks like this is due to a `setTimeout` of 600ms in the code: https://github.com/ionic-team/ionic-framework/blob/f81d18c6f9f1bce056afda1cac4cf6d6ace0a7ca/core/src/components/item-sliding/item-sliding.tsx#L361-L364.

**Comment by mrSingh007** [comment]
@liamdebeasi any workaround for that?

**Comment by kixes** [comment]
It has been almost one year. Do we have a solution or version with a fix for this yet?

**Comment by kixes** [comment]
I have created a flexible workaround:
```
  handleSlideBtn(event, func, ...parmas) {
    const slider = event.target.parentNode.parentNode.querySelector('ion-item');
    if ( !!slider.style.transform ) {
      func.apply(this, parmas);
    }
  }
```
so basically add the $event from click to determine if the slider is open or not based on the transform style. Then pass in your function that you would be calling for that button as the second parameter. Then pass in whatever parameters you would normally pass to the function. So in example:
```
<ion-item-option (click)="myFunc(parm1, parm2, parm3)">
```
would become:
```
<ion-item-option (click)="handleSlideBtn($event, myFunc, parm1, parm2, parm3)">
```
Note that this still doesn't run the click event on the ion-item if it exists but it does prevent the buttons from running while the slider is closed. The primary weak point in the handleSideBtn function is the slider variable, I wrote this under the assumption that this guide is followed strictly https://ionicframework.com/docs/api/item-sliding with no extra elements.
